### ISSUE TITLE
Test fail due to cursor on EditText

### ISSUE THREAD

**Issue by juanscigliano** [Karumi/Shot#41]
Hi,
Im trying to Screenshot my Login screen. My EditTex has focus when the activity start, and shows  the cursor over it. When I  execute the Screenshot Tests  this fail because some times the cursor is there some time not (native animation of cursor). What can i do?. (Remove the focus in not an option due to UX)
 Thanks.

**Comment by juanscigliano** [comment]
@pedrovgs Thanks for the answer, your approach make total sense.

**Comment by SUPERCILEX** [comment]
You can also recursively kill the cursors. Here's a full example for anyone wanting to copypasta stuff:

```kotlin
package support.bymason.kiosk.checkin.helpers

import android.app.Activity
import android.view.ViewGroup
import android.widget.TextView
import androidx.core.view.children
import androidx.test.core.app.ActivityScenario
import com.facebook.testing.screenshot.Screenshot

fun <T : Activity> ActivityScenario<T>.takeScreenshot() {
    lateinit var activity: Activity
    onActivity {
        activity = it
    }

    activity.hideCursors()
    Thread.sleep(500) // Wait for animations to finish

    // The automated screenshot naming is based off of stack trace analysis. We can't put this in
    // the onActivity lambda because that gets posted to the main thread.
    Screenshot.snapActivity(activity).record()
}

private fun Activity.hideCursors() {
    runOnUiThread { findViewById<ViewGroup>(android.R.id.content).hideCursors() }
}

private fun ViewGroup.hideCursors() {
    for (child in children) {
        if (child is ViewGroup) child.hideCursors()
        if (child is TextView) child.isCursorVisible = false
    }
}
```

**Comment by mecoFarid** [comment]
@SUPERCILEX Assumptions like `wait 500 ms then you should be good to go`  are the reason most tests fail and we stay like a stickman thinking about what the hell caused the issue

**Comment by mecoFarid** [comment]
> TLDR: You need to update your code to disable animations and the cursor just during the tests execution.

Rewriting the app :)
